target/linux/realtek/files-5.10/drivers/net/ethernet/rtl838x_eth.c

```diff
@@ -1811,7 +1811,7 @@ static int rtl930x_mdio_write_paged(struct mii_bus *bus, int mii_id, u16 page,
 	struct rtl838x_eth_priv *priv = bus->priv;
 	int err;
 
-	if (priv->sds_id[mii_id] >= 0)
+	if (priv->phy_is_internal[mii_id])
```

Comment:
This change should be folded into the change adding support for rtl930x.
And isn't that needed for the write function for rtl930x as well?

Reply:
For the read function it was already the case. I'll squash this tomorrow. I wanted to be sure that the change was seen.

Reply:
Probably it makes sense to fold-in the patch I've added fixing Clause-45 MDIO write on the RTL931x...